driver-core/src/main/com/mongodb/internal/connection/ClusterableServer.java

```diff
@@ -34,6 +35,15 @@
      */
     void invalidate(Throwable reason);
 
+    /**
+     * Invalidate the description of this server due to the passed in reason.
+     * @param reason the reason for invalidation.
+     * @param sessionContext the session context.
+     *
+     * @since 3.12
+     */
+    void invalidate(Throwable reason, SessionContext sessionContext);
```

Comment:
Does this need to be part of the interface?  Does it compile if you remove this?

Reply:
I have removed this from the interface.